myhoard/controller.py

```diff
@@ -1264,7 +1290,10 @@ def _process_removed_binlogs(self, binlogs):
                 stream.remove_binlogs(binlogs)
 
     def _purge_old_backups(self):
-        purgeable = [backup for backup in self.state["backups"] if backup["completed_at"]]
+        # wait for acquiring lock in case the controller got a request for refreshing backups.
+        # this way preserve_until and broken_at attributes are up to date before purging
+        with self.lock:
```

Comment:
ah true, this lock doesn't make sense anymore after I changed the way of marking preservation. Was more intended in case there was a race condition... but no concern anymore regards that.